source/common/stats/isolated_store_impl.cc

```diff
@@ -19,17 +19,28 @@ IsolatedStoreImpl::IsolatedStoreImpl(std::unique_ptr<SymbolTable>&& symbol_table
 }
 
 IsolatedStoreImpl::IsolatedStoreImpl(SymbolTable& symbol_table)
-    : alloc_(symbol_table), counters_([this](StatName name) -> CounterSharedPtr {
-        return alloc_.makeCounter(name, name, StatNameTagVector{});
+    : alloc_(symbol_table),
+      counters_([this](const TagUtility::TagStatNameJoiner& joiner,
+                       StatNameTagVectorOptConstRef tags) -> CounterSharedPtr {
+        return alloc_.makeCounter(joiner.nameWithTags(), joiner.tagExtractedName(),
+                                  tags ? tags->get() : StatNameTagVector{});
       }),
-      gauges_([this](StatName name, Gauge::ImportMode import_mode) -> GaugeSharedPtr {
-        return alloc_.makeGauge(name, name, StatNameTagVector{}, import_mode);
+      gauges_([this](const TagUtility::TagStatNameJoiner& joiner, StatNameTagVectorOptConstRef tags,
+                     Gauge::ImportMode import_mode) -> GaugeSharedPtr {
+        return alloc_.makeGauge(joiner.nameWithTags(), joiner.tagExtractedName(),
+                                tags ? tags->get() : StatNameTagVector{}, import_mode);
       }),
-      histograms_([this](StatName name, Histogram::Unit unit) -> HistogramSharedPtr {
-        return HistogramSharedPtr(new HistogramImpl(name, unit, *this, name, StatNameTagVector{}));
+      histograms_([this](const TagUtility::TagStatNameJoiner& joiner,
+                         StatNameTagVectorOptConstRef tags,
+                         Histogram::Unit unit) -> HistogramSharedPtr {
+        return {new HistogramImpl(joiner.nameWithTags(), unit, *this, joiner.tagExtractedName(),
+                                  tags ? tags->get() : StatNameTagVector{})};
       }),
-      text_readouts_([this](StatName name, TextReadout::Type) -> TextReadoutSharedPtr {
-        return alloc_.makeTextReadout(name, name, StatNameTagVector{});
+      text_readouts_([this](const TagUtility::TagStatNameJoiner& joiner,
+                            StatNameTagVectorOptConstRef tags,
+                            TextReadout::Type) -> TextReadoutSharedPtr {
+        return alloc_.makeTextReadout(joiner.nameWithTags(), joiner.tagExtractedName(),
+                                      tags ? tags->get() : StatNameTagVector{});
```

Comment:
Just to make this less verbose could you factor the conditional expression into a static local helper method with signature
`StatNameTagOrVector tagVectorFromOpt(StatNameTagVectorOptConstRef tags);`

Reply:
Done (also in the predecessor PR).